fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java

```diff
@@ -0,0 +1,98 @@
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.trees.plans.commands;
+
+import org.apache.doris.analysis.RecoverDbStmt;
+import org.apache.doris.analysis.StmtType;
+import org.apache.doris.catalog.Env;
+import org.apache.doris.common.ErrorCode;
+import org.apache.doris.common.ErrorReport;
+import org.apache.doris.common.UserException;
+import org.apache.doris.datasource.InternalCatalog;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.nereids.trees.plans.visitor.PlanVisitor;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.StmtExecutor;
+
+import com.google.common.base.Strings;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * recover database command
+ */
+public class RecoverDatabaseCommand extends Command implements ForwardWithSync {
+
+    public static final Logger LOG = LogManager.getLogger(RecoverDatabaseCommand.class);
+    private String dbName;
+    private long dbId = -1;
+    private String newDbName = "";
+
+    /**
+     * constructor
+     */
+
+    public RecoverDatabaseCommand(String dbName, long dbId, String newDbName) {
+        super(PlanType.RECOVER_DATABASE_COMMAND);
+        this.dbName = dbName;
+        this.dbId = dbId;
+        if (newDbName != null) {
+            this.newDbName = newDbName;
+        }
+    }
+
+    public String getDbName() {
+        return dbName;
+    }
+
+    public long getDbId() {
+        return dbId;
+    }
+
+    public String getNewDbName() {
+        return newDbName;
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws UserException {
+        if (Strings.isNullOrEmpty(dbName)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_WRONG_DB_NAME, dbName);
+        }
+
+        if (!Env.getCurrentEnv().getAccessManager()
+                .checkDbPriv(ConnectContext.get(), InternalCatalog.INTERNAL_CATALOG_NAME, dbName,
+                        PrivPredicate.ALTER_CREATE)) {
+            ErrorReport.reportAnalysisException(
+                    ErrorCode.ERR_DBACCESS_DENIED_ERROR, ctx.getQualifiedUser(), dbName);
+        }
+
+        RecoverDbStmt recoverStmt = new RecoverDbStmt(dbName, dbId, newDbName);
```

Comment:
pls call the new recoverDatabase(String dbName, long dbId, String newDbName) function, and remove dependence of  RecoverDbStmt